lib/AST/ASTContext.cpp

```diff
@@ -4289,6 +4289,9 @@ void DeclName::CompoundDeclName::Profile(llvm::FoldingSetNodeID &id,
 
 void DeclName::initialize(ASTContext &C, DeclBaseName baseName,
                           ArrayRef<Identifier> argumentNames) {
+  if (!baseName.isSpecial())
+    assert(baseName.getIdentifier() != C.getIdentifier("init"));
```

Comment:
Isn't this the very thing that will break with `` `init` ``?

Reply:
The assertion is only temporary while I work out some other issues.